Description: fix FTBFS
Author: Michael Stapelberg <stapelberg@debian.org>
Forwarded: https://github.com/snapcore/snapd/pull/4646
Last-Update: 2018-02-10

---

Index: snapd/cmd/snap-seccomp/main.go
===================================================================
--- snapd.orig/cmd/snap-seccomp/main.go
+++ snapd/cmd/snap-seccomp/main.go
@@ -20,8 +20,8 @@
 package main
 
 //#cgo CFLAGS: -D_FILE_OFFSET_BITS=64
-//#cgo pkg-config: --static --cflags libseccomp
-//#cgo LDFLAGS: -Wl,-Bstatic -lseccomp -Wl,-Bdynamic
+//#cgo pkg-config: libseccomp
+//#cgo LDFLAGS: -lseccomp
 //
 //#include <asm/ioctls.h>
 //#include <ctype.h>
