Index: snapd.mvo/interfaces/seccomp/backend_test.go
===================================================================
--- snapd.mvo.orig/interfaces/seccomp/backend_test.go
+++ snapd.mvo/interfaces/seccomp/backend_test.go
@@ -355,6 +355,8 @@ func (s *backendSuite) TestSocketcallIsA
 }
 
 func (s *backendSuite) TestSocketcallIsNotAddedWhenNotRequired(c *C) {
+	c.Skip("skipped until debian has check.v1 from 2018. Not() is broken in the old version")
+
 	restore := seccomp.MockRequiresSocketcall(func(string) bool { return false })
 	defer restore()
 
@@ -376,6 +378,8 @@ apps:
   `
 
 func (s *backendSuite) TestSystemKeyRetLogSupported(c *C) {
+	c.Skip("skipped until debian has check.v1 from 2018. Not() is broken in the old version")
+
 	restore := release.MockSecCompActions([]string{"allow", "errno", "kill", "log", "trace", "trap"})
 	defer restore()
 
@@ -396,6 +400,8 @@ func (s *backendSuite) TestSystemKeyRetL
 }
 
 func (s *backendSuite) TestSystemKeyRetLogUnsupported(c *C) {
+	c.Skip("skipped until debian has check.v1 from 2018. Not() is broken in the old version")
+
 	restore := release.MockSecCompActions([]string{"allow", "errno", "kill", "trace", "trap"})
 	defer restore()
 
