diff --git a/Make.rules b/Make.rules
index 903a5a4..5328063 100644
--- a/Make.rules
+++ b/Make.rules
@@ -81,7 +81,7 @@ endif
 	./cert-to-efi-sig-list -g $(MYGUID) $< $@
 
 getcert = $(shell if [ "$(1)" = "PK" -o "$(1)" = "KEK" ]; then echo "-c PK.crt -k PK.key"; else echo "-c KEK.crt -k KEK.key"; fi)
-getvar = $(shell if [ "$(1)" = "PK" -o "$(1)" = "KEK" ]; then echo $(1); else echo db; fi)
+getvar = $(shell if [ "$(1)" = "PK" -o "$(1)" = "KEK" ]; then echo $(1); elif [ "$(1)" = ms-kek ]; then echo KEK; else echo db; fi)
 
 %.auth: %.esl PK.crt KEK.crt sign-efi-sig-list
 	./sign-efi-sig-list $(call getcert,$*) $(call getvar,$*) $< $@
