diff --git a/UpdateVars.c b/UpdateVars.c
index 2d21563..00027cb 100644
--- a/UpdateVars.c
+++ b/UpdateVars.c
@@ -28,15 +28,19 @@ efi_main (EFI_HANDLE image, EFI_SYSTEM_TABLE *systab)
 	EFI_GUID *owner;
 	CHAR16 **variables;
 	EFI_GUID **owners;
+	CHAR16 *variables_dbt[] = { L"PK", L"KEK", L"db", L"dbx", L"dbt", L"MokList" , NULL};
+	EFI_GUID *owners_dbt[] = { &GV_GUID, &GV_GUID, &SIG_DB, &SIG_DB, &SIG_DB, &MOK_OWNER };
+	CHAR16 *variables_nodbt[] = { L"PK", L"KEK", L"db", L"dbx", L"MokList" , NULL};
+	EFI_GUID *owners_nodbt[] = { &GV_GUID, &GV_GUID, &SIG_DB, &SIG_DB, &MOK_OWNER };
 
 	InitializeLib(image, systab);
 
 	if (GetOSIndications() & EFI_OS_INDICATIONS_TIMESTAMP_REVOCATION) {
-		variables = (CHAR16 *[]){ L"PK", L"KEK", L"db", L"dbx", L"dbt", L"MokList" , NULL};
-		owners = (EFI_GUID *[]){ &GV_GUID, &GV_GUID, &SIG_DB, &SIG_DB, &SIG_DB, &MOK_OWNER };
+		variables = variables_dbt;
+		owners = owners_dbt;
 	} else {
-		variables = (CHAR16 *[]){ L"PK", L"KEK", L"db", L"dbx", L"MokList" , NULL};
-		owners = (EFI_GUID *[]){ &GV_GUID, &GV_GUID, &SIG_DB, &SIG_DB, &MOK_OWNER };
+		variables = variables_nodbt;
+		owners = owners_nodbt;
 	}
 
 	status = argsplit(image, &argc, &ARGV);
