diff --git b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.current.fstab a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.current.fstab
index aadd564c33..d55e3f9269 100644
--- b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.current.fstab
+++ a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.current.fstab
@@ -1,3 +1,4 @@
+# This file is similar to -1-, except that attached content has different revision.
 tmpfs / tmpfs x-snapd.origin=rootfs 0 0
 tmpfs /usr/share/secureboot tmpfs x-snapd.synthetic,x-snapd.needed-by=/usr/share/secureboot/potato,mode=0755,uid=0,gid=0 0 0
 /usr/share/secureboot/updates /usr/share/secureboot/updates none rbind,x-snapd.synthetic,x-snapd.needed-by=/usr/share/secureboot/potato,x-snapd.detach 0 0
diff --git b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.desired.fstab a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.desired.fstab
index 98d9c0ef53..2de313d056 100644
--- b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.desired.fstab
+++ a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/5-initially-connected-then-content-refreshed.desired.fstab
@@ -1,2 +1,7 @@
+# This file is almost the same as 1-initially-connected.desired.fstab, which is good.
+# The only difference, and one that we expect, is the revision of the attached content snap.
+# The revision changes from x1 to x2 as the content is refreshed.
 /snap/test-snapd-layout/x2/attached-content /usr/share/secureboot/potato none rbind,rw,x-snapd.origin=layout 0 0
+# This is the content entry. Notice that it is after the layout but the user
+# expectation is that the content shows up through the layout entry.
 /snap/test-snapd-content/x2 /snap/test-snapd-layout/x2/attached-content none bind,ro 0 0
