diff --git b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.current.fstab a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.current.fstab
index e6a6f386ab..161f422d67 100644
--- b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.current.fstab
+++ a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.current.fstab
@@ -1,3 +1,4 @@
+# The order is no longer broken.
 tmpfs / tmpfs x-snapd.origin=rootfs 0 0
 tmpfs /usr/share/secureboot tmpfs x-snapd.synthetic,x-snapd.needed-by=/usr/share/secureboot/potato,mode=0755,uid=0,gid=0 0 0
 /usr/share/secureboot/updates /usr/share/secureboot/updates none rbind,x-snapd.synthetic,x-snapd.needed-by=/usr/share/secureboot/potato,x-snapd.detach 0 0
diff --git b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.desired.fstab a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.desired.fstab
index 7df3de1c3b..0b004668dc 100644
--- b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.desired.fstab
+++ a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/2-after-disconnect.desired.fstab
@@ -1 +1,3 @@
+# This is the layout entry. Snapd orders layouts before any content entries.
 /snap/test-snapd-layout/x2/attached-content /usr/share/secureboot/potato none rbind,rw,x-snapd.origin=layout 0 0
+# The content entry is now gone.
