diff --git b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/6-initially-connected-then-app-refreshed.desired.fstab a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/6-initially-connected-then-app-refreshed.desired.fstab
index b49b4096d7..f4d4734242 100644
--- b/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/6-initially-connected-then-app-refreshed.desired.fstab
+++ a/cmd/snap-update-ns/testdata/usr-share-secureboot-potato/6-initially-connected-then-app-refreshed.desired.fstab
@@ -1,2 +1,3 @@
+# The desired file is the same as -1-, apart from revision numbers, and correct.
 /snap/test-snapd-layout/x3/attached-content /usr/share/secureboot/potato none rbind,rw,x-snapd.origin=layout 0 0
 /snap/test-snapd-content/x1 /snap/test-snapd-layout/x3/attached-content none bind,ro 0 0
