diff --git a/usr/lib/systemd/system/populate-writable.service b/usr/lib/systemd/system/populate-writable.service
index 30d9ae3..c914015 100644
--- a/usr/lib/systemd/system/populate-writable.service
+++ b/usr/lib/systemd/system/populate-writable.service
@@ -3,12 +3,12 @@ OnFailure=emergency.target
 OnFailureJobMode=replace-irreversibly
 Before=initrd-cleanup.target
 After=run-mnt-ubuntu\x2dseed.mount
-After=run-mnt-ubuntu\x2ddata.mount
+After=run-mnt-data.mount
 After=run-mnt-base.mount
 After=sysroot.mount
 After=sysroot-writable.mount
 Requires=run-mnt-ubuntu\x2dseed.mount
-Requires=run-mnt-ubuntu\x2ddata.mount
+Requires=run-mnt-data.mount
 Requires=run-mnt-base.mount
 Requires=sysroot.mount
 Requires=sysroot-writable.mount
diff --git "a/usr/lib/systemd/system/run-mnt-ubuntu\\x2ddata.mount.wants/sysroot-writable.mount" "b/usr/lib/systemd/system/run-mnt-ubuntu\\x2ddata.mount.wants/sysroot-writable.mount"
deleted file mode 120000
index 0d3b832..0000000
--- "a/usr/lib/systemd/system/run-mnt-ubuntu\\x2ddata.mount.wants/sysroot-writable.mount"
+++ /dev/null
@@ -1 +0,0 @@
-../sysroot-writable.mount
\ No newline at end of file
diff --git a/usr/lib/systemd/system/sysroot-writable.mount b/usr/lib/systemd/system/sysroot-writable.mount
index 8d614a5..1006ff7 100644
--- a/usr/lib/systemd/system/sysroot-writable.mount
+++ b/usr/lib/systemd/system/sysroot-writable.mount
@@ -1,15 +1,15 @@
 [Unit]
 Before=initrd-cleanup.target
-After=run-mnt-ubuntu\x2ddata.mount
+After=run-mnt-data.mount
 After=run-mnt-base.mount
 After=sysroot.mount
 After=the-tool.service
-Requires=run-mnt-ubuntu\x2ddata.mount
+Requires=run-mnt-data.mount
 Requires=run-mnt-base.mount
 Requires=sysroot.mount
 Requires=the-tool.service
 Wants=populate-writable.service
 [Mount]
-What=/run/mnt/ubuntu-data
+What=/run/mnt/data
 Type=none
 Options=bind
